### PR TITLE
fix(codeium-vim): port mappings to `astrocore` and fix lazy loading event

### PR DESCRIPTION
**Pull request by manuuurino**

### CODE REVIEW

File: lua/astrocommunity/completion/codeium-vim/init.lua [hunk 1]
```diff
@@ -1,17 +1,45 @@
 return {
   "Exafunction/codeium.vim",
-  event = "User AstroFile",
```
Comment (manuuurino) [inline lua/astrocommunity/completion/codeium-vim/init.lua:3]: In my testing, ive noticed that this event no longer works? Cant explain why, havent done the research on it, kinda tired rn. @mehalter Is it a potential bug from AstroNvim? Could look sometime later if you arent intrested.

Comment (Uzaaft) [reply lua/astrocommunity/completion/codeium-vim/init.lua:3]: Lazy loading with vim pugins is wonky. We should just explicitly disable it with
`lazy = false` instead.

Comment (manuuurino) [reply lua/astrocommunity/completion/codeium-vim/init.lua:3]: Would rather go with the `BufEnter` event and it works, thats how they used it in their [examples](https://github.com/Exafunction/codeium.vim/blob/7a01ffb1e6538fe634f294908a50a7b680039284/README.md#-lazy), also ive implemented it already in this pr.

Comment (Uzaaft) [reply lua/astrocommunity/completion/codeium-vim/init.lua:3]: Sure if you've tested it lets go for that.

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**